train_agent.py

```diff
@@ -63,16 +65,63 @@ def main(steps, gpu, imagefile, boxfile):
 
     logging.basicConfig(level=logging.INFO, stream=sys.stdout, format='')
 
+    step_hooks = []
+    if tensorboard:
+        timestr = time.strftime("%Y%m%d-%H%M%S")
+        agentClassName = agent.__class__.__name__[:10]
+        writer = SummaryWriter("tensorboard/tensorBoard_exp_" + timestr + "_" + agentClassName)
+        step_hooks = [TensorBoardLoggingStepHook(writer)]
+
     chainerrl.experiments.train_agent_with_evaluation(
         agent, env,
         steps=steps,  # Train the agent for 5000 steps
         eval_n_runs=10,  # 10 episodes are sampled for each evaluation
         max_episode_len=50,  # Maximum length of each episodes
         eval_interval=500,  # Evaluate the agent after every 100 steps
-        outdir='result')  # Save everything to 'result' directory
+        outdir='result', # Save everything to 'result' directory
+        step_hooks=step_hooks)
 
     agent.save('agent')
 
+class TensorBoardLoggingStepHook(chainerrl.experiments.StepHook):
+    def __init__(self, summary_writer):
+        self.summary_writer = summary_writer
+        return
+
+    def __call__(self, env, agent, step):
+        step_count = agent.t
+
+        self.summary_writer.add_scalar('average_q', agent.average_q, step_count)
+
+        self.summary_writer.add_scalar('average_loss', agent.average_loss, step_count)
+
+        last_action_name = env.action_set[agent.last_action].__name__
+        debug_image = self.get_debug_image_from_environment_image(image = env.episode_image.copy(),
+                                                                  bbox = env.bbox,
+                                                                  last_action_name = last_action_name)
+
+        # Prepare the image for further use in rb_chainer's add_image() method
+        debug_image = np.array(debug_image, np.float32) / 255
+        debug_image = debug_image.transpose((2, 0, 1))
+
+        self.summary_writer.add_image('episode_image_debug', debug_image, step_count)
+        return
+
+    def get_debug_image_from_environment_image(self, image, bbox, last_action_name):
+        image = image.copy()
+
+        debug_image = Image.new(mode='RGB',
+                                size=(image.width, image.height + 16),
+                                color=0)
+        debug_image.paste(image)
+
+        draw = ImageDraw.Draw(debug_image)
+        draw.rectangle(bbox.tolist(), outline=(255, 255, 255))
+        draw.text(xy=(2, image.height + 2),
+                  text='Last action: %s' % last_action_name,
+                  fill='white')
```

Comment:
@bakoe Add font and font size to avoid possible errors caused by different/changing defaults.